kroxylicious/src/main/java/io/kroxylicious/proxy/config/tls/TrustStore.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.config.tls;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.security.KeyStore;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.netty.handler.ssl.SslContextBuilder;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+/**
+ * A {@link TrustProvider} backed by a Java Truststore.
+ *
+ * @param storeFile             location of a key store, or reference to a PEM file containing both private-key/certificate/intermediates.
+ * @param storePasswordProvider password used to protect the key store.
```

Comment:
nit: Possibly worth being explicit here that passwords are optional.

Reply:
Fixed.